manager-parent/manager-v2/src/main/java/com/redhat/service/smartevents/manager/v2/workers/resources/AbstractWorker.java

```diff
@@ -0,0 +1,264 @@
+            try {
+                updated = createDependencies(work, managedResource);
+            } catch (Exception e) {
+                LOGGER.error(String.format("Failed to create dependencies for resource of type '%s' with id '%s'.", work.getType(), id), e);
+                // Something has gone wrong. We need to retry.
+                workManager.rescheduleAfterFailure(work);
+            } finally {
+                if (!isProvisioningComplete(updated)) {
+                    workManager.reschedule(work);
+                }
+            }
+        } else if (OperationType.DELETE.equals(managedResource.getOperation().getType())) {
+            try {
+                updated = deleteDependencies(work, managedResource);
+            } catch (Exception e) {
+                LOGGER.info(String.format("Failed to delete dependencies for resource of type '%s' with id '%s'.", work.getType(), id), e);
+                // Something has gone wrong. We need to retry.
+                workManager.rescheduleAfterFailure(work);
+            } finally {
+                if (!isDeprovisioningComplete(updated)) {
+                    workManager.reschedule(work);
+                }
+            }
+        }
+
+        return updated;
+    }
+
+    public abstract String getId(Work work);
+
+    @Transactional
+    protected T load(String id) {
+        return getDao().findByIdWithConditions(id);
+    }
+
+    @Transactional
+    protected T persist(T managedResource) {
+        return getDao().getEntityManager().merge(managedResource);
+    }
+
+    protected boolean areRetriesExceeded(Work w, ManagedResource managedResource) {
+        boolean areRetriesExceeded = w.getAttempts() > maxRetries;
+        if (areRetriesExceeded) {
+            LOGGER.error(
+                    "Max retry attempts exceeded trying to create dependencies for '{}' [{}].",
+                    managedResource.getName(),
+                    managedResource.getId());
+        }
+        return areRetriesExceeded;
+    }
+
+    protected boolean isTimeoutExceeded(Work work, ManagedResource managedResource) {
+        boolean isTimeoutExceeded = ZonedDateTime.now(ZoneOffset.UTC).minusSeconds(timeoutSeconds).isAfter(work.getSubmittedAt());
+        if (isTimeoutExceeded) {
+            LOGGER.error(
+                    "Timeout exceeded trying to create dependencies for '{}' [{}].",
+                    managedResource.getName(),
+                    managedResource.getId());
+        }
+        return isTimeoutExceeded;
+    }
+
+    protected T recordError(Work work) {
+        String managedResourceId = getId(work);
+        T managedResource = load(managedResourceId);
+
+        markFailedConditions(managedResource);
+
+        return persist(managedResource);
+    }
+
+    public abstract ManagedResourceV2DAO<T> getDao();
+
+    protected abstract T createDependencies(Work work, T managedResource);
```

Comment:
Code style alert!

It might have been nicer to keep the `deleteDependencies(..)` [method](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/832d9988897aec262033347ad3ba717239a96f53/manager-parent/manager-v2/src/main/java/com/redhat/service/smartevents/manager/v2/workers/resources/AbstractWorker.java#L238) closer to here (than add the new code inbetween them).